### ISSUE TITLE
Unterminated attribute value in a template results in cryptic AOT compilation errors with cli (or no errors with ngc)

### ISSUE THREAD

**Issue by IgorMinar** [angular/angular#20902]
## I'm submitting a...

<pre><code>
[x] Bug report  
</code></pre>

## Current behavior

Given a template:
```
hello <div class=">world</div>
```
@angular/cli reports very cryptic error and @angular/compiler-cli doesn't report any errors at all. JIT compiler reports correct error message.

## Expected behavior
I'd like to get an error that tells me that I have unterminated attribute value.

This works correctly in JIT: http://plnkr.co/edit/x7wVVkf8VROCPSlkszJS?p=preview

## Minimal reproduction of the problem with instructions

```
ng new foo
cd foo
cat 'hello<div class=">world</div>' > ./src/app/app.component.html 
```

then use ngc to build:

```
yarn ngc -p src/tsconfig.app.json 
```

And you get no error:
```
yarn run v1.3.2
$ /Users/iminar/Dev/ng-hq/client/node_modules/.bin/ngc -p src/tsconfig.app.json
✨  Done in 2.94s.
```

then use cli+aot to build:
```
yarn ng build --aot
```
and you get uninformative error from the guts of webpack:
```
yarn run v1.3.2
$ ng build --aot
Date: 2017-12-09T00:25:57.363Z                                                          
Hash: 06f14a38832d43ca634b
Time: 3809ms
chunk {inline} inline.bundle.js, inline.bundle.js.map (inline) 5.83 kB [entry] [rendered]
chunk {main} main.bundle.js, main.bundle.js.map (main) 1.08 MB [initial] [rendered]
chunk {polyfills} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 199 kB [initial] [rendered]
chunk {styles} styles.bundle.js, styles.bundle.js.map (styles) 11.4 kB [initial] [rendered]

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './app/app.module.ngfactory' in '/Users/iminar/Dev/ng-hq/client/src'
resolve './app/app.module.ngfactory' in '/Users/iminar/Dev/ng-hq/client/src'
  using description file: /Users/iminar/Dev/ng-hq/client/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /Users/iminar/Dev/ng-hq/client/package.json (relative path: ./src)
    using description file: /Users/iminar/Dev/ng-hq/client/package.json (relative path: ./src/app/app.module.ngfactory)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        /Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        /Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        /Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory.js doesn't exist
      as directory
        /Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory doesn't exist
[/Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory]
[/Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory.ts]
[/Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory.js]
[/Users/iminar/Dev/ng-hq/client/src/app/app.module.ngfactory]
 @ ./src/main.ts 3:0-66
 @ multi ./src/main.ts
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Environment

<pre><code>
Angular version: 5.1.0
CLI: 1.6.0
</code></pre>

**Comment by chuckjaz** [comment]
Fixed by #20846

**Comment by IgorMinar** [comment]
Thanks!
